src/braket/circuits/instruction.py

```diff
@@ -81,6 +83,24 @@ def target(self) -> QubitSet:
         """
         return self._target
 
+    def adjoint(self) -> Instruction:
+        """Returns an Instruction with the adjoint of this instruction's own operator
+
+        This operation only works on Gate operators.
```

Comment:
Is this true? You seem to have a switch statement for CompilerDirective.

Reply:
Whoops, should be "and CompilerDirectives"